Plugins/ExternalTexture/Source/ExternalTexture_OpenGL.cpp

```diff
@@ -18,21 +130,108 @@ namespace Babylon::Plugins
         Impl(Graphics::TextureT);
         void Update(Graphics::TextureT);
 
+        ~Impl()
+        {
+            if (m_textureID)
+            {
+                glDeleteTextures(1, &m_textureID);
+            }
+        }
+
         uintptr_t Ptr() const
         {
-            throw std::runtime_error{"not implemented"};
+            return m_textureID;
         }
 
     private:
-        void GetInfo(Graphics::TextureT, Info&)
+        int countMipMapLevelsExcludingBase(GLuint textureID)
+        {
+            // Bind the texture
+            glBindTexture(GL_TEXTURE_2D, textureID);
+
+            int width, height;
+            int mipLevel = 1; // Start from level 1 (excluding level 0)
+            int mipCount = 0;
+
+            // Query the width and height of each mipmap level
+            while (true)
+            {
+                // Get the texture width and height at the current mipmap level
+                glGetTexLevelParameteriv(GL_TEXTURE_2D, mipLevel, GL_TEXTURE_WIDTH, &width);
+                glGetTexLevelParameteriv(GL_TEXTURE_2D, mipLevel, GL_TEXTURE_HEIGHT, &height);
+
+                // If the width or height is 0, the mipmap level does not exist
+                if (width == 0 && height == 0)
+                {
+                    break; // No more mipmap levels
+                }
+
+                // Increment the mipmap level and count
+                mipLevel++;
+                mipCount++;
+            }
+
+            // Return the number of mipmap levels excluding level 0
+            return mipCount;
+        }
+
+        void GetInfo(Graphics::TextureT textureID, Info& info)
         {
-            throw std::runtime_error{"not implemented"};
+            glBindTexture(GL_TEXTURE_2D, textureID); // Bind the texture
```

Comment:
Probably should store the previously bound texture and restore it at the end instead of resetting to zero.